packages/cactus-plugin-ledger-connector-iroha/src/main/typescript/plugin-ledger-connector-iroha.ts

```diff
@@ -0,0 +1,665 @@
+        connector: this,
+        logLevel: this.options.logLevel,
+      };
+      const endpoint = new GetPrometheusExporterMetricsEndpointV1(opts);
+      endpoints.push(endpoint);
+    }
+    this.endpoints = endpoints;
+    return endpoints;
+  }
+
+  public getPackageName(): string {
+    return `@hyperledger/cactus-plugin-ledger-connector-iroha`;
+  }
+
+  public async getConsensusAlgorithmFamily(): Promise<
+    ConsensusAlgorithmFamily
+  > {
+    return ConsensusAlgorithmFamily.Authority;
+  }
+  public async hasTransactionFinality(): Promise<boolean> {
+    const currentConsensusAlgorithmFamily = await this.getConsensusAlgorithmFamily();
+
+    return consensusHasTransactionFinality(currentConsensusAlgorithmFamily);
+  }
+  //...  spread
+  //pass in the params
+  public async transact(
+    req: RunTransactionRequest, //string + array<any>
+  ): Promise<RunTransactionResponse> {
+    const adminPriv =
+      "f101537e319568c765b2cc89698325604991dca57b9716b58016b253506cab70";
+    const commandService = new CommandService(
+      "localhost:50051",
+      grpc.credentials.createInsecure(),
+    );
+    const queryService = new QueryService(
+      "localhost:50051",
+      grpc.credentials.createInsecure(),
+    );
+    const commandOptions = {
+      privateKeys: [adminPriv],
+      creatorAccountId: "admin@test",
+      quorum: 1,
+      commandService: commandService,
+      timeoutLimit: 5000,
+    };
+    const commandOptions2 = {
+      privateKeys: [adminPriv],
+      creatorAccountId: "admin@test",
+      quorum: 2,
+      commandService: commandService,
+      timeoutLimit: 5000,
+    };
+    const queryOptions = {
+      privateKey: adminPriv,
+      creatorAccountId: "admin@test",
+      queryService: queryService,
+      timeoutLimit: 5000,
+    };
+    console.log(req);
+
+    switch (req.commandName) {
+      case IrohaCommand.CreateAccount: {
+        try {
+          const response = await commands.createAccount(commandOptions, {
+            accountName: req.params[0],
+            domainId: req.params[1],
+            publicKey: req.params[2],
+          });
+          return { transactionReceipt: response };
+        } catch (err) {
+          throw new Error(err);
+        }
+      }
+      case IrohaCommand.SetAccountDetail: {
+        try {
+          const response = await commands.setAccountDetail(commandOptions, {
+            accountId: req.params[0],
+            key: req.params[1],
+            value: req.params[2],
+          });
+          return { transactionReceipt: response };
+        } catch (err) {
+          throw new Error(err);
+        }
+      }
+      case IrohaCommand.CompareAndSetAccountDetail: {
+        try {
+          const response = await commands.compareAndSetAccountDetail(
+            commandOptions,
+            {
+              accountId: req.params[0],
+              key: req.params[1],
+              value: req.params[2],
+              oldValue: req.params[3],
```

Comment:
Thank you for pointing out this! Eventually the request will be an interface, so the fields will automatically be optional instead of required by the array.

Reply:
Thank you for pointing out this glitch! Unfortunately, the Javascript library refused to accept the `checkEmpty` parameter:
![image](https://user-images.githubusercontent.com/10954157/129434326-27c9c274-aeaa-4544-9b27-fdac37665dbf.png)
![image](https://user-images.githubusercontent.com/10954157/129434416-1c537a1d-5745-4fa8-bf45-3d16fad172b1.png)

I will add that into one of the "to-be-improved" list.